### PR TITLE
Fix #1167 : Replace apiextensions.k8s.io/v1beta1 by  apiextensions.k8s.io/v1

### LINKED ISSUE #1167: CustomResourceDefinition uses v1beta1 API
### Description
C.f. [Deprecated API Migration Guide - CustomResourceDefinition](https://kubernetes.io/docs/reference/using-api/deprecation-guide/#customresourcedefinition-v122)

### Info

* Eclipse JKube version : 1.5.1
* Maven version (`mvn -v`) : 3.8.3
* Kubernetes version : 1.22.3
* If it's a bug, how to reproduce :
  * Add a crd to your project
  * Run 'mvn k8s:deploy', this will show the following error :
     ```
     Execution default-cli of goal org.eclipse.jkube:kubernetes-maven-plugin:1.5.1:apply failed: Failure executing: GET at: https://127.0.0.1:59558/apis/apiextensions.k8s.io/v1beta1/customresourcedefinitions. Message: the server could not find the requested resource. Received status: Status(apiVersion=v1, code=404, details=StatusDetails(causes=[], group=null, kind=null, name=null, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=the server could not find the requested resource, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=NotFound, status=Failure, additionalProperties={}). -> [Help 1]
    ```

### PR DESCRIPTION
**Pull request by guiliguili**
## Description
Fix #1167 

Replace apiextensions.k8s.io/v1beta1 by  apiextensions.k8s.io/v1 for CustomResourceDefinition

## Type of change
 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [X] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [X] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

## Additional notes
When running `mvn verify had to ignore the following tests because it was failing on file assertions due to absolute/relative path resolution issue. It does not seem to be related to the fix :
* AssemblyFileSetUtilsProcessAssemblyFileSetTest
* AssemblyManagerCreateDockerTarArchiveTest
* JibBuildServiceBuildIntegrationTest

### CODE REVIEW

**Comment by rohanKanojia** [comment]
Welcome, Thanks a lot for contributing to Eclipse JKube :+1:,

In order to contribute to Eclipse, you would need to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca). Once you've created an eclipse account you would need to update your commit message with a `Signed-off-by:` footer. You can do it like this:
```
# (Optional): Configure Git User and Email
git config user.name "Your Name"
git config user.email "your_email@example.com"

# Amend your previous commit with signed off footer
git commit --amend --signoff
```

**Comment by guiliguili** [comment]
> Welcome, Thanks a lot for contributing to Eclipse JKube 👍,
> 
> In order to contribute to Eclipse, you would need to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca). Once you've created an eclipse account you would need to update your commit message with a `Signed-off-by:` footer. You can do it like this:
> 
> ```
> # (Optional): Configure Git User and Email
> git config user.name "Your Name"
> git config user.email "your_email@example.com"
> 
> # Amend your previous commit with signed off footer
> git commit --amend --signoff
> ```

Done

**Comment by rohanKanojia** [comment]
strange, ECA check still seems to be failing. Are you sure that the email you used for Eclipse Account is the same as one used in commit message?

**Comment by rohanKanojia** [comment]
Umm, is it possible to sign ECA with your work account and update the commit with that email?

**Comment by guiliguili** [comment]
> Umm, is it possible to sign ECA with your work account and update the commit with that email?

I updated my Eclipse primary email to my work email and re-executed the ECA. Should be fine now.

**Review by rohanKanojia — APPROVED** [review]

File: jkube-kit/common/src/test/resources/util/resource-util/list-with-standard-template-and-cr-resources.yml [hunk 1]
```diff
@@ -62,7 +62,7 @@ items:
       - name: limits_memory
         value: "512Mi"
       - name: parameter_with_no_value
-  - apiVersion: apiextensions.k8s.io/v1beta1
+  - apiVersion: apiextensions.k8s.io/v1
```
Comment (manusa) [inline jkube-kit/common/src/test/resources/util/resource-util/list-with-standard-template-and-cr-resources.yml:65]: This test shouldn't be affected

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Review by rohanKanojia — APPROVED** [review]

**Pull request merged**